salt/metalk8s/orchestrate/upgrade/init.sls

```diff
@@ -0,0 +1,71 @@
+Execute the upgrade prechecks:
+  salt.runner:
+    - name: state.orchestrate
+    - mods:
+      - metalk8s.orchestrate.upgrade.precheck
+    - saltenv: {{ saltenv }}
+
+Upgrade etcd cluster:
+  salt.runner:
+    - name: state.orchestrate
+    - mods:
+      - metalk8s.orchestrate.upgrade.etcd
+    - saltenv: {{ saltenv }}
+    - require:
+      - salt: Execute the upgrade prechecks
+
+{%- set dest_version = pillar.orchestrate.dest_version %}
+{%- set cp_nodes = salt.metalk8s.minions_by_roles('master') | sort %}
+{%- set other_nodes = pillar.metalk8s.nodes.keys() | difference(cp_nodes) | sort %}
+
+{%- for node in cp_nodes + other_nodes %}
+
+  {%- set kubeconfig = "/etc/kubernetes/admin.conf" %}
+  {%- set context = "kubernetes-admin@kubernetes" %}
+
+Set node {{ node }} version to {{ dest_version }}:
+  metalk8s_kubernetes.node_label_present:
+    - name: metalk8s.scality.com/version
+    - node: {{ node }}
+    - value: "{{ dest_version }}"
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - require:
+      - salt: Upgrade etcd cluster
+  {%- if previous_node is defined %}
+      - salt: Deploy node {{ previous_node }}
+  {%- endif %}
+
+Deploy node {{ node }}:
+  salt.runner:
+    - name: state.orchestrate
+    - mods:
+      - metalk8s.orchestrate.deploy_node
+    - saltenv: {{ saltenv }}
+    - pillar:
+        orchestrate:
+          node_name: {{ node }}
+    - require:
+      - metalk8s_kubernetes: Set node {{ node }} version to {{ dest_version }}
+    - require_in:
+      - salt: Deploy Kubernetes objects
+
+  {#- Ugly but needed since we have jinja2.7 (`loop.previtem` added in 2.10) #}
+  {%- set previous_node = node %}
+
+{%- endfor %}
+
+Deploy Kubernetes objects:
+  salt.runner:
+    - name: state.orchestrate
+    - mods:
```

Comment:
We should bundle this in some SLS which can then be reused by the `bootstrap` playbook as well.

Reply:
Yes also thinking about this. I will